### PR TITLE
Rename and deprecate Zeebe classes

### LINKED ISSUE #19488: Create a new `CamundaClientBuilder`
Create a new `CamundaClientBuilder`, `ZeebeClientBuilder` still can be used by users,

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

With this PR all the `Zeebe....` classes are replaced or deprecated in favor of more general classes. 

## Related issues

closes #19485 
closes #19488

### CODE REVIEW

File: clients/java/pom.xml
Comment (tmetzke) [inline clients/java/pom.xml:300]: 🔧 You should be able to move the configuration from the `execution` level to the `plugin` level. You can then just overwrite the `modelPackage` in he configuration section of the "old-rest" execution. That removes duplication.

File: clients/java/src/main/java/io/camunda/zeebe/client/api/JsonMapper.java
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/JsonMapper.java]: ❓ Did this happen by mistake?

File: clients/java/src/main/java/io/camunda/zeebe/client/api/worker/JobClient.java
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/worker/JobClient.java]: ❓ Did this happen by mistake?

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java:77]: 🔧 Those changes need to stay, I think. We adjusted the model, so this won't work otherwise with the `Zeebe...` classes anymore.

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientImpl.java
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientImpl.java:186]: ❓ Why did we change this for the Zeebe client originally? I think this needs to stay. Was it due to the renaming of the artifact?

Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientImpl.java:186]: Yes, I changed it while changing the artifact name. Good catch 👍

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/http/ApiResponseConsumer.java
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/http/ApiResponseConsumer.java:46]: 🔧 Those changes were a fix, please leave them in 👍

Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/http/ApiResponseConsumer.java:46]: This change is not in the stable branch then 👀

Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/http/ApiResponseConsumer.java:46]: ❌

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java
Comment (tmetzke) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java:77]: As synced: let's rollback those changes as it is here now and also use V1 REST YAML to generate the "old-rest" data model.

File: clients/java/pom.xml
Comment (tmetzke) [reply clients/java/pom.xml:300]: Let's also use V1 REST YAML to generate the "old-rest" models.

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java
Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java:77]: As synced (PT.2): We don't need to rollback those changes

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/http/ApiResponseConsumer.java
Comment (tmetzke) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/http/ApiResponseConsumer.java:46]: For reference: those changes are needed when communicating with an 8.6 cluster to appropriately receive error messages. We need those changes in the 8.6 client in Zeebe and Camunda classes. This is currently not backported to 8.5 since it's meant to be compatible with 8.5 clusters, if I am not mistaken.

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java
Comment (tmetzke) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java:77]: Back and forth, love it 😆

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java:135]: ❓ Chnages by accident, I guess?

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProvider.java
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:148]: ❓ Chnages by accident, I guess?

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/response/DeploymentEventImpl.java
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/response/DeploymentEventImpl.java:38]: ❓ Chnages by accident, I guess?

File: clients/java/src/test/java/io/camunda/zeebe/client/OAuthCredentialsProviderTest.java
Comment (tmetzke) [inline clients/java/src/test/java/io/camunda/zeebe/client/OAuthCredentialsProviderTest.java:363]: ❓ Chnages by accident, I guess?

Comment (nicpuppa) [reply clients/java/src/test/java/io/camunda/zeebe/client/OAuthCredentialsProviderTest.java:363]: Those changes are due to the copy/paste from stable/8.5

File: clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java
Comment (tmetzke) [inline clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java:89]: 🔧 Those changes were introduced as a fix here: https://github.com/camunda/camunda/pull/19635
Should we rather backport the changes and keep them here as well?

File: operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/DataGenerator.java
Comment (tmetzke) [inline operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/DataGenerator.java:66]:
```suggestion
   * ZeebeClient must not be reused between different test fixtures, as this may be different
```

File: clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java
Comment (nicpuppa) [reply clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java:89]: @mustafadagher I'm tagging you because you made those changes, what do you think ?

File: operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/DataGenerator.java
Comment (tmetzke) [inline operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/DataGenerator.java:69]: ❓ So, should we rather move to the CamundaClient then?

File: operate/qa/integration-tests/src/test/java/io/camunda/operate/util/OperateTester.java
Comment (tmetzke) [inline operate/qa/integration-tests/src/test/java/io/camunda/operate/util/OperateTester.java:261]: 🔧 If we name this CamundaClient, we should also use it. Or stick with the Zeebe client for now.

File: operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/DataGenerator.java
Comment (nicpuppa) [reply operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/DataGenerator.java:69]: Nope, I will rollback the changes in the doc. I think operate devs should decide if they want to switch and using camunda client

File: operate/qa/integration-tests/src/test/java/io/camunda/operate/util/OperateTester.java
Comment (nicpuppa) [reply operate/qa/integration-tests/src/test/java/io/camunda/operate/util/OperateTester.java:261]: I think IntellJ modify those classes, so I'm going to rollback them

File: clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java
Comment (mustafadagher) [reply clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java:89]: The changes mainly reside in test, and they were aimed to fix flaky tests. it's safe to backport them if needed

**Comment by mustafadagher** [comment]
@nicpuppa I see you have been manually backporting some merged PRs manually here. I'm curious why did you choose to manually commit those changes instead of rebasing the branch on main?

**Comment by nicpuppa** [comment]
With this PR we introduce some structural changes to the java client:
- In `io.camunda.client` there is the new Camunda Java Client with all the new feature and fix that should be delivered in the next release
- In `io.camunda.zeebe.client` there is the Zeebe Java Client (updated at the last stable version)

Why we took this decision ? Because we want to enable users to use the new client (and the new features) without changing all the codebase. Users can use the new functionalities (using the new `CamundaClient`) but they can still use the old `ZeebClient` without any problem.

In this way, once we want to deprecate the old `ZeebeClient`, we can just delete the package easily. In the meantime we need to take care of some duplication of the code, but I'm going to create a workflow on GitHub as a reminder

If you have any question, please let me know. Once this PR is merged. I will send a communication on slack 👍 

cc. @mustafadagher

**Review by megglos — APPROVED** [review]
Approving given previous comments have been addressed and to unblock follow-up changes and avoid conflicts.

@tmetzke please check this still if you consider it needed

**Pull request merged**
